### PR TITLE
Create your own Enumerable

### PR DESCRIPTION
**Pull request by Recillah-Khamala**
### **Create your own Enumerable**

In this milestone, my partner and I have-:

- Created a class MyList that has an instance variable @list.
- Implemented in MyList a method #each that yields successive members of @list and uses the MyEnumerable module.
- Created a module MyEnumerable that implements the following methods:
  - all?
  - any?
  - filter
- Verified our solution.

### CODE REVIEW

File: my_enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+module MyEnumerable
+  def all?
+    result = []
+    @list.each do |element|
+      result.push(element) if yield element
+    end
+    puts result.length == @list.length
```
Comment (DammyShittu) [inline my_enumerable.rb]: - [ ] Typically methods are supposed to `return` a result not just print it. Consider a scenario where we would like to chain methods or reuse the result of the method in another method. Kindly remove `puts` here.

  ```
  result.length == @list.length
  ```

  If we wish to print the result of this method we can call the method.
  ```
  puts list.all? { |e| e < 5 }
  ```

File: my_enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  def any?
+    @list.each do |element|
+      if yield element
+        puts true
+        return true
+      end
+    end
+    puts false
+  end
```
Comment (DammyShittu) [inline my_enumerable.rb]: - [ ] Kindly remove these `puts` as well

  ```
  def any?
    @list.each do |element|
      if yield element
        return true
      end
    end
    false
  end
  ```

File: my_enumerable.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  def filter
+    result = []
+    @list.each do |element|
+      result.push(element) if yield element
+    end
+    print result
```
Comment (DammyShittu) [inline my_enumerable.rb]: - [ ] Kindly remove `print` from this statement.

  ```
  result
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Recillah-Khamala @RichardChileya,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: my_enumerable.rb (hunk 2 above)
Comment (RichardChileya) [reply my_enumerable.rb]: resolved

File: my_enumerable.rb (hunk 3 above)
Comment (thecodechaser) [inline my_enumerable.rb]: - [ ] As requested by the previous, Please remove the `puts` and `prints` command inside the functions and just return the value. In the commit history, I can see you fixed the issues and then reverted back. Please make sure you implement the requested changes by the previous reviewer.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @Recillah-Khamala, and @RichardChileya 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ismailco — APPROVED** [review]
### Hi @Recillah-Khamala,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**